pkg/moov/bank_account.go

```diff
@@ -145,3 +168,72 @@ func (c Client) MicroDepositConfirm(ctx context.Context, accountID string, bankA
 		return resp.Error()
 	}
 }
+
+// CreatePlaidLink creates a new bank account for the given customer account using Plaid processor_token
+func (c Client) CreateBankAccountWithPlaid(ctx context.Context, accountID string, plaid Plaid) (*BankAccount, error) {
+	payload := BankAccountPayload{
+		Plaid: &plaid,
+	}
+	resp, err := c.CallHttp(ctx,
+		Endpoint(http.MethodPost, pathBankAccounts, accountID),
+		AcceptJson(),
+		JsonBody(payload))
+	if err != nil {
+		return nil, err
+	}
+
+	switch resp.Status() {
+	case StatusCompleted:
+		return CompletedObjectOrError[BankAccount](resp)
+	case StatusStateConflict:
+		return nil, ErrDuplicateBankAccount
+	default:
+		return nil, resp.Error()
+	}
+}
+
+// CreateBankAccountWithPlaidLink creates a new bank account for the given customer account using Plaid public_token
+func (c Client) CreateBankAccountWithPlaidLink(ctx context.Context, accountID string, plaid PlaidLink) (*BankAccount, error) {
```

Comment:
I think we should get this down to the one function and make the argument a Option function to set mx or plaid etc.. So it would look like `CreateBankAccount(ctx, accountID, WithPlainLink())`

Reply:
Agreed. That would be nicer to use.